### PR TITLE
Log when page not found

### PR DESCRIPTION
**Pull request by furkansenharputlu**
Example log:

``` 
time="Jan 13 16:07:06" level=error msg="404 page not found" request="GET /non-existing-api/get HTTP/1.1" 
```

Fixes https://github.com/TykTechnologies/tyk/issues/2468

### CODE REVIEW

File: gateway/api_loader.go [hunk 1]
```diff
@@ -676,10 +676,12 @@ func loadApps(specs []*APISpec) {
...
 	muxer := &proxyMux{}
 
-	globalConf := config.Global()
 	r := mux.NewRouter()
-	muxer.setRouter(globalConf.ListenPort, "", r)
+	r.NotFoundHandler = http.HandlerFunc(muxer.handle404)
+
+	globalConf := config.Global()
 	if globalConf.ControlAPIPort == 0 {
+		muxer.setRouter(globalConf.ListenPort, "", r)
```
Comment (gernest) [inline gateway/api_loader.go]: Why did you move this?  I think you are breaking the logic here.

So, `r` is bound on `globalConf.ListenPort` by default, if we don't have a separate `globalConf.ControlAPIPort` we will load control endpoints on `r` . When that control port is provided then we bind control endpoints there.

Please leave `muxer.setRouter(globalConf.ListenPort, "", r)` above the `if` condition as it was before.

Comment (furkansenharputlu) [reply gateway/api_loader.go]: If it enter if, there is nothing change.
if it enters else statement, It is overriden in else statement. 

So I think it doesn't break.

Comment (furkansenharputlu) [reply gateway/api_loader.go]: Let me make it more simple.

**Comment by gernest** [comment]
@furkansenharputlu please only leave the changes related to `404` page. Leave everything else as it was. The `404` changes already look good. You can make any unrelated changes in a separate PR( With more details on why you made them) .

**Comment by furkansenharputlu** [comment]
Yes, you're actually right.  However, the current code forces me to make redundant changes while implementing 404. Therefore, I will open a separate PR and after merging it. I will make the 404 changes on top of it.

**Comment by furkansenharputlu** [comment]
@gernest The other PRs are merged. I think this is ready to go. Can you please review?

File: gateway/proxy_muxer.go [hunk 2]
```diff
@@ -148,6 +148,14 @@ func (m *proxyMux) setRouter(port int, protocol string, router *mux.Router) {
 	}
 }
 
+func (m *proxyMux) handle404(w http.ResponseWriter, r *http.Request) {
+	requestMeta := fmt.Sprintf("%s %s %s", r.Method, r.URL.Path, r.Proto)
+	log.WithField("request", requestMeta).Error("404 page not found")
+
+	w.WriteHeader(http.StatusNotFound)
+	fmt.Fprintf(w, "404 page not found")
```
Comment (gernest) [inline gateway/proxy_muxer.go]: lets use standard library here `fmt.Fprint(w, http.StatusText(http.StatusNotFound))` . Use `fmt.Fprint` because we are not formatting the string.

File: gateway/proxy_muxer_test.go [hunk 3]
```diff
@@ -289,3 +291,18 @@ func TestHTTP_custom_ports(t *testing.T) {
...
+	defer g.Close()
+
+	BuildAndLoadAPI(func(spec *APISpec) {
+		spec.Proxy.ListenPath = "/existing"
+		spec.UseKeylessAccess = true
+	})
+
+	_, _ = g.Run(t, []test.TestCase{
+		{Path: "/existing", Code: http.StatusOK},
+		{Path: "/nonexisting", Code: http.StatusNotFound, BodyMatch: "404 page not found"},
```
Comment (gernest) [inline gateway/proxy_muxer_test.go]: use `http.StatusText(http.StatusNotFound)` on `BodyMatch`

**Comment by asoorm** [comment]
@furkansenharputlu This should be configurable - we should be able to turn it on and off. Can get extremely noisy if we just blindly log everything...
